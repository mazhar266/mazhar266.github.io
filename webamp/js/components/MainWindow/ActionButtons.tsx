import React from "react";
import { connect } from "react-redux";
import { previous, play, pause, stop, next } from "../../actionCreators";
import { Dispatch } from "../../types";

interface DispatchProps {
  previous(): void;
  play(): void;
  pause(): void;
  stop(): void;
  next(): void;
}

const ActionButtons = (props: DispatchProps) => (
  <div className="actions">
    <div id="previous" onClick={props.previous} title="Previous Track" />
    <div id="play" onClick={props.play} title="Play" />
    <div id="pause" onClick={props.pause} title="Pause" />
    <div id="stop" onClick={props.stop} title="Stop" />
    <div id="next" onClick={props.next} title="Next Track" />
  </div>
);

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => {
  return {
    previous: () => dispatch(previous()),
    play: () => dispatch(play()),
    pause: () => dispatch(pause()),
    stop: () => dispatch(stop()),
    next: () => dispatch(next()),
  };
};

export default connect(
  null,
  mapDispatchToProps
)(ActionButtons);
