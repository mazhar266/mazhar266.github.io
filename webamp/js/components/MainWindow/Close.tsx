import React from "react";
import { connect } from "react-redux";
import ClickedDiv from "../ClickedDiv";

import { close } from "../../actionCreators";
import { Dispatch } from "../../types";

interface DispatchProps {
  onClick: () => void;
}

const Close = ({ onClick }: DispatchProps) => (
  <ClickedDiv id="close" onClick={onClick} title="Close" />
);

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => {
  return { onClick: () => dispatch(close()) };
};

export default connect(
  null,
  mapDispatchToProps
)(Close);
