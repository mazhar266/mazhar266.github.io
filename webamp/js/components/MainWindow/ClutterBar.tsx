import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import { SET_FOCUS, UNSET_FOCUS } from "../../actionTypes";
import { toggleDoubleSizeMode } from "../../actionCreators";
import { AppState, Dispatch } from "../../types";
import OptionsContextMenu from "../OptionsContextMenu";
import ContextMenuTarget from "../ContextMenuTarget";

interface StateProps {
  doubled: boolean;
}

interface DispatchProps {
  handleMouseDown(): void;
  handleMouseUp(): void;
}

const ClutterBar = (props: StateProps & DispatchProps) => (
  <div id="clutter-bar">
    <ContextMenuTarget bottom handle={<div id="button-o" />}>
      <OptionsContextMenu />
    </ContextMenuTarget>
    <div id="button-a" />
    <div id="button-i" />
    <div
      title={"Toggle Doublesize Mode"}
      id="button-d"
      className={classnames({ selected: props.doubled })}
      onMouseUp={props.handleMouseUp}
      onMouseDown={props.handleMouseDown}
    />
    <div id="button-v" />
  </div>
);

const mapStateToProps = (state: AppState): StateProps => ({
  doubled: state.display.doubled,
});

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  handleMouseDown: () => dispatch({ type: SET_FOCUS, input: "double" }),
  handleMouseUp: () => {
    dispatch(toggleDoubleSizeMode());
    dispatch({ type: UNSET_FOCUS });
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ClutterBar);
