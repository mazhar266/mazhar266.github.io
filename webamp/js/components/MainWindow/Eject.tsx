import React from "react";
import { connect } from "react-redux";

import * as Actions from "../../actionCreators";
import { Dispatch } from "../../types";

interface DispatchProps {
  openMediaFileDialog(): void;
}

const Eject = (props: DispatchProps) => (
  <div id="eject" onClick={props.openMediaFileDialog} title="Open File(s)" />
);

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  openMediaFileDialog: () => dispatch(Actions.openMediaFileDialog()),
});

export default connect(
  null,
  mapDispatchToProps
)(Eject);
