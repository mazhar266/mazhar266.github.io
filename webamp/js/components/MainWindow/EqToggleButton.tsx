import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import { getWindowOpen } from "../../selectors";
import { toggleWindow } from "../../actionCreators";
import { AppState } from "../../types";

interface StateProps {
  windowOpen: boolean;
}

interface DispatchProps {
  handleClick(e: React.MouseEvent<HTMLDivElement>): void;
}

type Props = StateProps & DispatchProps;

const EqToggleButton = (props: Props) => (
  <div
    id="equalizer-button"
    className={classnames({ selected: props.windowOpen })}
    onClick={props.handleClick}
    title="Toggle Graphical Equalizer"
  />
);

const mapStateToProps = (state: AppState): StateProps => ({
  windowOpen: getWindowOpen(state)("equalizer"),
});

const mapDispatchToProps = {
  handleClick: () => toggleWindow("equalizer"),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EqToggleButton);
