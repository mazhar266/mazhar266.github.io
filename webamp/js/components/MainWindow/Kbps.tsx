import React from "react";
import { connect } from "react-redux";

import CharacterString from "../CharacterString";
import { AppState } from "../../types";
import * as Selectors from "../../selectors";

interface StateProps {
  kbps: string | null;
}

const Kbps = (props: StateProps) => (
  <div id="kbps">
    <CharacterString>{props.kbps || ""}</CharacterString>
  </div>
);

function mapStateToProps(state: AppState): StateProps {
  return { kbps: Selectors.getKbps(state) };
}

export default connect(mapStateToProps)(Kbps);
