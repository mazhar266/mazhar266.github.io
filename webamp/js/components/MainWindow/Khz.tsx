import React from "react";
import { connect } from "react-redux";

import CharacterString from "../CharacterString";
import { AppState } from "../../types";
import * as Selectors from "../../selectors";

interface StateProps {
  khz: string | null;
}

const Khz = (props: StateProps) => (
  <div id="khz">
    <CharacterString>{props.khz || ""}</CharacterString>
  </div>
);

function mapStateToProps(state: AppState): StateProps {
  return { khz: Selectors.getKhz(state) };
}

export default connect(mapStateToProps)(Khz);
