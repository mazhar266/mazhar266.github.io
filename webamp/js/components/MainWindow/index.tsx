import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";
import { WINDOWS, MEDIA_STATUS, LOAD_STYLE } from "../../constants";
import {
  loadFilesFromReferences,
  toggleMainWindowShadeMode,
  scrollVolume,
  loadMedia,
} from "../../actionCreators";
import { getWindowShade } from "../../selectors";

import DropTarget from "../DropTarget";
import MiniTime from "../MiniTime";

import ClickedDiv from "../ClickedDiv";
import ContextMenuTarget from "../ContextMenuTarget";
import Visualizer from "../Visualizer";
import ActionButtons from "./ActionButtons";
import MainBalance from "./MainBalance";
import Close from "./Close";
import ClutterBar from "./ClutterBar";
import MainContextMenu from "./MainContextMenu";
import Eject from "./Eject";
import EqToggleButton from "./EqToggleButton";
import PlaylistToggleButton from "./PlaylistToggleButton";
import Kbps from "./Kbps";
import Khz from "./Khz";
import Marquee from "./Marquee";
import MonoStereo from "./MonoStereo";
import Position from "./Position";
import Repeat from "./Repeat";
import Shade from "./Shade";
import Minimize from "./Minimize";
import Shuffle from "./Shuffle";
import Time from "./Time";
import MainVolume from "./MainVolume";

import "../../../css/main-window.css";
import {
  MediaStatus,
  WindowId,
  AppState,
  Dispatch,
  FilePicker,
} from "../../types";
import FocusTarget from "../FocusTarget";

interface StateProps {
  focused: WindowId | null;
  loading: boolean;
  doubled: boolean;
  mainShade: boolean;
  llama: boolean;
  working: boolean;
  status: MediaStatus | null;
}

interface DispatchProps {
  loadFilesFromReferences(files: FileList): void;
  scrollVolume(e: React.WheelEvent<HTMLDivElement>): void;
  toggleMainWindowShadeMode(): void;
  loadMedia(e: React.DragEvent<HTMLDivElement>): void;
}

interface OwnProps {
  analyser: AnalyserNode;
  filePickers: FilePicker[];
}

type Props = StateProps & DispatchProps & OwnProps;

export class MainWindow extends React.Component<Props> {
  render() {
    const {
      focused,
      loading,
      doubled,
      mainShade,
      llama,
      status,
      working,
      filePickers,
    } = this.props;

    const className = classnames({
      window: true,
      play: status === MEDIA_STATUS.PLAYING,
      stop: status === MEDIA_STATUS.STOPPED,
      pause: status === MEDIA_STATUS.PAUSED,
      selected: focused === WINDOWS.MAIN,
      shade: mainShade,
      draggable: true,
      loading,
      doubled,
      llama,
    });

    return (
      <React.StrictMode>
        <DropTarget
          id="main-window"
          className={className}
          handleDrop={this.props.loadMedia}
          onWheel={this.props.scrollVolume}
        >
          <FocusTarget windowId={WINDOWS.MAIN}>
            <div
              id="title-bar"
              className="selected draggable"
              onDoubleClick={this.props.toggleMainWindowShadeMode}
            >
              <ContextMenuTarget
                id="option-context"
                bottom
                handle={<ClickedDiv id="option" title="Winamp Menu" />}
              >
                <MainContextMenu filePickers={filePickers} />
              </ContextMenuTarget>
              {mainShade && <MiniTime />}
              <Minimize />
              <Shade />
              <Close />
            </div>
            <div className="status">
              <ClutterBar />
              {!working && <div id="play-pause" />}
              <div
                id="work-indicator"
                className={classnames({ selected: working })}
              />
              <Time />
            </div>
            <Visualizer
              // @ts-ignore Visualizer is not typed yet
              analyser={this.props.analyser}
            />
            <div className="media-info">
              <Marquee />
              <Kbps />
              <Khz />
              <MonoStereo />
            </div>
            <MainVolume />
            <MainBalance />
            <div className="windows">
              <EqToggleButton />
              <PlaylistToggleButton />
            </div>
            <Position />
            <ActionButtons />
            <Eject />
            <div className="shuffle-repeat">
              <Shuffle />
              <Repeat />
            </div>
            <a
              id="about"
              target="blank"
              href="https://webamp.org/about"
              title="About"
            />
          </FocusTarget>
        </DropTarget>
      </React.StrictMode>
    );
  }
}

const mapStateToProps = (state: AppState): StateProps => {
  const {
    media: { status },
    display: { loading, doubled, llama, working },
    windows: { focused },
  } = state;
  return {
    mainShade: Boolean(getWindowShade(state)("main")),
    status,
    loading,
    doubled,
    llama,
    working,
    focused,
  };
};

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => {
  return {
    loadFilesFromReferences: (files: FileList) =>
      dispatch(loadFilesFromReferences(files)),
    toggleMainWindowShadeMode: () => dispatch(toggleMainWindowShadeMode()),
    scrollVolume: (e: React.WheelEvent<HTMLDivElement>) =>
      dispatch(scrollVolume(e)),
    loadMedia: (e: React.DragEvent<HTMLDivElement>) =>
      dispatch(loadMedia(e, LOAD_STYLE.PLAY)),
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps,
  null,
  { forwardRef: true }
)(MainWindow);
