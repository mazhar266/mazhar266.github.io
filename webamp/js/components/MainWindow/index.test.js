import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import mockGetStore from "../../__mocks__/storeMock";
import media from "../../__mocks__/mediaMock";
import { SET_SKIN_DATA } from "../../actionTypes";

import MainWindow from "./index";

const canvasMockify = require("canvas-mock");

function createNodeMock(element) {
  if (element.type === "canvas") {
    const Canvas = window.document.createElement("canvas");
    canvasMockify(Canvas); // mock canvas functions required by Phaser.js are added
    return Canvas;
  }
  return null;
}

describe("MainWindow", () => {
  let store;
  beforeEach(() => {
    store = mockGetStore();
    store.dispatch({ type: SET_SKIN_DATA, data: {} });
  });

  it("renders to snapshot", () => {
    const options = { createNodeMock };
    const tree = renderer
      .create(
        <Provider store={store}>
          <MainWindow mediaPlayer={media} />
        </Provider>,
        options
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
