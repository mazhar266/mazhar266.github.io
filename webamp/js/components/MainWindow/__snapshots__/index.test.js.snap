// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainWindow renders to snapshot 1`] = `
<div
  className="window stop selected draggable"
  id="main-window"
  onDragEnter={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  onWheel={[Function]}
>
  <div
    onFocus={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    tabIndex={-1}
  >
    <div
      className="selected draggable"
      id="title-bar"
      onDoubleClick={[Function]}
    >
      <div
        id="option-context"
      >
        <div
          className="handle"
          onClick={[Function]}
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <div
            className=""
            id="option"
            onMouseDown={[Function]}
            title="Winamp Menu"
          />
        </div>
      </div>
      <div
        className=""
        id="minimize"
        onClick={[Function]}
        onMouseDown={[Function]}
        title="Minimize"
      />
      <div
        className=""
        id="shade"
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
        title="Toggle Windowshade Mode"
      />
      <div
        className=""
        id="close"
        onClick={[Function]}
        onMouseDown={[Function]}
        title="Close"
      />
    </div>
    <div
      className="status"
    >
      <div
        id="clutter-bar"
      >
        <div>
          <div
            className="handle"
            onClick={[Function]}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <div
              id="button-o"
            />
          </div>
        </div>
        <div
          id="button-a"
        />
        <div
          id="button-i"
        />
        <div
          className=""
          id="button-d"
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          title="Toggle Doublesize Mode"
        />
        <div
          id="button-v"
        />
      </div>
      <div
        id="play-pause"
      />
      <div
        className=""
        id="work-indicator"
      />
      <div
        className="countdown"
        id="time"
        onClick={[Function]}
      >
        <div
          className="digit digit-0"
          id="minute-first-digit"
        />
        <div
          className="digit digit-0"
          id="minute-second-digit"
        />
        <div
          className="digit digit-0"
          id="second-first-digit"
        />
        <div
          className="digit digit-0"
          id="second-second-digit"
        />
      </div>
    </div>
    <canvas
      height={32}
      id="visualizer"
      onClick={[Function]}
      style={
        Object {
          "height": 16,
          "width": 76,
        }
      }
      width={152}
    />
    <div
      className="media-info"
    >
      <div
        className="text"
        id="marquee"
        onMouseDown={[Function]}
        title="Song Title"
      >
        <div
          style={
            Object {
              "transform": "translateX(0px)",
              "whiteSpace": "nowrap",
              "willChange": "transform",
            }
          }
        >
          <span
            className=" character character-119"
          >
            W
          </span>
          <span
            className=" character character-105"
          >
            i
          </span>
          <span
            className=" character character-110"
          >
            n
          </span>
          <span
            className=" character character-97"
          >
            a
          </span>
          <span
            className=" character character-109"
          >
            m
          </span>
          <span
            className=" character character-112"
          >
            p
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-50"
          >
            2
          </span>
          <span
            className=" character character-46"
          >
            .
          </span>
          <span
            className=" character character-57"
          >
            9
          </span>
          <span
            className=" character character-49"
          >
            1
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
          <span
            className=" character character-32"
          >
             
          </span>
        </div>
      </div>
      <div
        id="kbps"
      />
      <div
        id="khz"
      />
      <div
        className="mono-stereo"
      >
        <div
          className=""
          id="stereo"
        />
        <div
          className=""
          id="mono"
        />
      </div>
    </div>
    <div
      id="volume"
      style={
        Object {
          "backgroundPosition": "0 -315px",
        }
      }
    >
      <input
        max="100"
        min="0"
        onChange={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        step="1"
        title="Volume Bar"
        type="range"
        value={78}
      />
    </div>
    <input
      id="balance"
      max="100"
      min="-100"
      onChange={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      step="1"
      style={
        Object {
          "backgroundPosition": "0 -0px",
        }
      }
      title="Balance"
      type="range"
      value={0}
    />
    <div
      className="windows"
    >
      <div
        className="selected"
        id="equalizer-button"
        onClick={[Function]}
        title="Toggle Graphical Equalizer"
      />
      <div
        className="selected"
        id="playlist-button"
        onClick={[Function]}
        title="Toggle Playlist Editor"
      />
    </div>
    <input
      className="left"
      id="position"
      max="100"
      min="0"
      onChange={[Function]}
      onInput={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      step="1"
      title="Seeking Bar"
      type="range"
      value={0}
    />
    <div
      className="actions"
    >
      <div
        id="previous"
        onClick={[Function]}
        title="Previous Track"
      />
      <div
        id="play"
        onClick={[Function]}
        title="Play"
      />
      <div
        id="pause"
        onClick={[Function]}
        title="Pause"
      />
      <div
        id="stop"
        onClick={[Function]}
        title="Stop"
      />
      <div
        id="next"
        onClick={[Function]}
        title="Next Track"
      />
    </div>
    <div
      id="eject"
      onClick={[Function]}
      title="Open File(s)"
    />
    <div
      className="shuffle-repeat"
    >
      <div
        onContextMenu={[Function]}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          className=""
          id="shuffle"
          onClick={[Function]}
          title="Toggle Shuffle"
        />
      </div>
      <div
        onContextMenu={[Function]}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          className=""
          id="repeat"
          onClick={[Function]}
          title="Toggle Repeat"
        />
      </div>
    </div>
    <a
      href="https://webamp.org/about"
      id="about"
      target="blank"
      title="About"
    />
  </div>
</div>
`;
